src/components/Hero/Hero.js

```diff
@@ -0,0 +1,227 @@
+                        move: 'run',
+                        posX: getNextPos(5)
+                    }
+                    updateHeroState(newHeroState)
+                }
+            },
+        }
+    }
+
+    const touchActionsMap = {
+        touchStart: (event) => {
+            if (!gameOnPause) {
+                setTouchState({y: event.touches[0].clientY, x: event.touches[0].clientX})
+            }
+        },
+
+        touchMove: (event) => {
+            if (!gameOnPause && touchState !== "done") {
+                if (heroState.move !== "sit" && touchState.y - event.touches[0].clientY > 100) {
+                    keyActionsMap.keydown.up()
+                    setTouchState("done")
+                } else if (heroState.move !== "sit" && touchState.y - event.touches[0].clientY < -100) {
+                    keyActionsMap.keydown.down()
+                    setTouchState("done")
+                } else if (heroState.move !== "sit" && heroState.move !== "jump") {
+                    if (touchState.x - event.touches[0].clientX < -100) {
+                        keyActionsMap.keydown.right()
+                    } else if (touchState.x - event.touches[0].clientX > 100) {
+                        keyActionsMap.keydown.left()
+                    }
+                }
+            }
+        },
+
+        touchEnd: () => {
+            if (!gameOnPause) {
+                if (heroState.move === "sit") {
+                    keyActionsMap.keyup.down()
+                } else if (heroState.move === "run") {
+                    updateHeroState({move: 'stand'})
+                }
+            }
+        }
+    }
+
+    const updateHeroState = (state) => {
+        setHeroState(Object.assign({}, heroState, state));
+    }
+
+    const playSound = (sound) => {
+        if (heroData[sound]) {
+            heroDom.current.querySelector(`audio#${soundMap[sound]}`).play();
+        }
+    }
+
+    const getNextPos = (val) => {
+        if (heroState.posX + val > frameWidth) {
+            return 0
+        } else if (heroState.posX + val < 0) {
+            return frameWidth
+        } else {
+            return heroState.posX + val
+        }
+    }
+
+    useKeyPress(keyActionsMap.keydown, "keydown", !gameOnPause)
+    useKeyPress(keyActionsMap.keyup, "keyup")
+    useEventListener(touchActionsMap.touchStart, "touchstart")
+    useEventListener(touchActionsMap.touchMove, "touchmove")
+    useEventListener(touchActionsMap.touchEnd, "touchend")
+
+    useEffect(() => {
+        for (let sound in soundMap) {
+            if (heroData[sound]) {
+                const soundEffect = heroDom.current.querySelector(`audio#${soundMap[sound]}`);
+                soundEffect.muted = soundMuted;
+                soundEffect.volume = soundVolume / 100;
+            }
+        }
+    }, [heroData, soundMuted, soundVolume])
+
+    useEffect(() => {
+        updateHeroState({
+            size: heroData.sizes.default
+        })
+    }, [heroData])
```

Comment:
Don't ignore hooks warnings, add `updateHeroState` to a dependency array and wrap it in useCallback.